src/swapd/swap_state.rs

```diff
@@ -715,20 +715,31 @@ fn attempt_transition_to_init_maker(
             // send maker commit message to counter-party
             runtime.log_trace(format!("sending peer MakerCommit msg {}", &local_commit));
             runtime.send_peer(event.endpoints, PeerMsg::MakerCommit(local_commit.clone()))?;
-            match swap_role {
-                SwapRole::Bob => Ok(Some(SwapStateMachine::BobInitMaker(BobInitMaker {
-                    local_commit,
-                    local_params,
-                    funding_address: funding_address.unwrap(),
-                    remote_commit: commit,
-                    wallet,
-                    reveal: None,
-                }))),
-                SwapRole::Alice => Ok(Some(SwapStateMachine::AliceInitMaker(AliceInitMaker {
-                    local_params,
-                    remote_commit: commit,
-                    wallet,
-                }))),
+            match (swap_role, wallet.clone()) {
+                (SwapRole::Bob, Wallet::Bob(BobState { funding_tx, .. })) => {
+                    Ok(Some(SwapStateMachine::BobInitMaker(BobInitMaker {
+                        local_commit,
+                        local_params,
+                        funding_address: funding_tx.get_address().ok().unwrap(),
+                        remote_commit: commit,
+                        wallet,
+                        reveal: None,
+                    })))
+                }
+                (SwapRole::Alice, Wallet::Alice(_)) => {
+                    Ok(Some(SwapStateMachine::AliceInitMaker(AliceInitMaker {
+                        local_params,
+                        remote_commit: commit,
+                        wallet,
+                    })))
+                }
+                _ => {
+                    runtime.log_error(format!(
+                        "Invalid swap role {} for wallet {:?}",
```

Comment:
Implement format strings on Wallet enum, no need to debug display its entire content.

Reply:
ddb110cc